shared/src/main/kotlin/com/egm/stellio/shared/model/CompactedEntity.kt

```diff
@@ -67,9 +120,9 @@ private fun simplifyAttribute(value: Map<String, Any>, transformationParameters:
 }
 
 fun CompactedEntity.toFilteredLanguageProperties(languageFilter: String): CompactedEntity =
-    this.mapValues { (_, value) ->
-        applyAttributeTransformation(
-            value,
+    this.mapValues { entry ->
+        applyAttributeTransformationWithParameters(
+            entry,
             mapOf(QUERY_PARAM_LANG to languageFilter),
             ::filterLanguageProperty,
             ::filterMultiInstanceLanguageProperty
```

Comment:
This way you can remove applyAttributeTransformationWithParameters
```suggestion
   mapOf(QUERY_PARAM_LANG to languageFilter).let {
        additionalParam ->  this.mapValues { entry ->
        applyAttributeTransformation(
            entry,
            { filterLanguageProperty(it,additionalParam)},
            { filterMultiInstanceLanguageProperty(it,additionalParam) },
        )
```
(I can't add a suggestion who include line 130 so the suggestion will not work because of a missing "}".)